### ISSUE TITLE
Android TV Support and addition to TV Play Store?

### ISSUE THREAD

**Issue by Pentaphon** [anilbeesetti/nextplayer#322]
I'd like to be able to use this player on my Android TV device and for it to be on the Android TV Play Store. Can you add support for using this player on Android TV devices with full remote control support?

**Comment by darkj2k** [comment]
Same request.

**Comment by Learner00100** [comment]
@anilbeesetti yes much need one

**Comment by markg85** [comment]
Same request!
I suspect it's easy to enable usage on android/google tv, that's just a few manifest changes. But making it work with remote control input might require some more work.

**Comment by markg85** [comment]
Just wanted to notify you folks of this.
I just tried installing NextPlayer on a chromecast. It's a bit tricky but here's how to do it:

1. Make sure you're a developer in chromecast (for help, [an article](https://allaboutchromecast.com/enable-developer-options-in-chromecast-with-google-tv/) by tapping the os build version ~10x rapidly.
2. Connect to your chromecast using ADB over wifi (must be on the same network). You essentially connect to your chromecast using `adb connect <ip>:5555` (yes, the 5555 port), when you make this connection the first time you have to authenticate it in the popup that appears on the chromecast

And now you can do a streaming sideload of the NextPlayer apk using a command like: `adb -s <chromecast ip>:5555 install <apk file>`

If you got this far, you have the app installed.
But you won't find it in the main android gui because the flags for chromecast aren't enabled in this APK. You can still start it, but you have to navigate to your system apps menu. Browse all apps and there you'll find NextPlayer. Click it and it gives you a menu of options to choose from, including open.

That's all there is to it, not that complex right? ;)
It works surprisingly well!

**Comment by Pentaphon** [comment]
@anilbeesetti this should be labeled as a feature, not a bug

**Comment by wyjek** [comment]
I would love to see Android TV support too.

For anyone wondering - it is possible to sideload it, but the interface is unusable by a remote, you need to connect a physical mouse and keyboard or use an app like Zank Remote.

Chromecast support would be great too.

**Comment by Pentaphon** [comment]
> For anyone wondering - it is possible to sideload it, but the interface is unusable by a remote, you need to connect a physical mouse and keyboard or use an app like Zank Remote.

Yes. We need Next Player to work as an external player on TV devices. This is easily the top requested feature next to background playback.

**Comment by anilbeesetti** [comment]
I will try to prioritize this in the upcoming releases

**Comment by serrq** [comment]
Android TV support yeah!

**Comment by long76** [comment]
> I will try to prioritize this in the upcoming releases

you can apply #1196 for now and mark ATV support like experimental?

**Comment by long76** [comment]
`0.12.3` with my fix. universal apk inside.
[nextplayer-v0.12.3-universal-ATV.zip](https://github.com/user-attachments/files/18280409/nextplayer-v0.12.3-universal-ATV.zip)
if you don't trust me:
install JDK 11, download [apktool](https://github.com/iBotPeaches/Apktool/releases/tag/v2.10.0), [build-tools 33.0.3](https://androidsdkmanager.com/build_tools.html) android
```shell
java -jar apktool_2.10.0.jar d nextplayer-v0.12.3-universal.apk
edit AndroidManifest.xml like in MR
java -jar apktool_2.10.0.jar b nextplayer-v0.12.3-universal
now we need sign apk
keytool -genkey -v -keystore my.keystore -keyalg RSA -keysize 4096 -validity 10000 -alias app
.\Android\sdk\build-tools\33.0.2\zipalign.exe -p 4 .\nextplayer-v0.12.3-universal\dist\nextplayer-v0.12.3-universal.apk nextplayer-aligned.apk
.\Android\sdk\build-tools\33.0.2\apksigner.bat sign --ks-key-alias app --ks .\my.keystore .\nextplayer-aligned.apk
```
and that's all)

P.S. happy new year and holidays)

**Comment by notvlexz** [comment]
0.14.1 is latest version where we can navigate the menu on android tv.  
Looks like we still don't have an official tv version sadly, but I'd appreciate if the creator just makes the menu at least navigateable on tv for the latest apk versions :(
Just in the meantime until we get a tv version pleaseee

**Comment by anilbeesetti** [comment]
Added in latest version (0.17.0)
